validator_client/src/keystore/kdf.rs

```diff
@@ -0,0 +1,165 @@
+use crypto::sha2::Sha256;
+use crypto::{hmac::Hmac, mac::Mac, pbkdf2, scrypt};
+use rand::prelude::*;
+use serde::{de, Deserialize, Serialize, Serializer};
+use std::default::Default;
+
+// TODO: verify size of salt
+const SALT_SIZE: usize = 32;
```

Comment:
From some ***quick*** reading, RFC 2898 (PBKDF2-SHA-256) specifies an "eight octet" salt (64 bits) whilst RFC 7914 (scrypt) does not specify the salt size.

It seems the test vectors have 32-byte salts, though.

Reply:
RFC 2898 specifies that the salt should be **atleast** 64 bits.

I don't see any harm in reducing to the minimum recommended value. Its only while deriving the key that we use the  `SALT_SIZE` constant. We will still be able to read keystores with larger salt values (for tests etc). 

Also, `scrypt` uses `pbkdf2` under the hood, so I guess we should have ultimately have similar salt sizes for both.

Reply:
Are you suggesting to drop `SALT_SIZE` to 8 bytes (64 bits)?

Reply:
I checked the other implementations (Go, Python, Typescript). All are using 32 byte salts. I suggest we keep it the same.